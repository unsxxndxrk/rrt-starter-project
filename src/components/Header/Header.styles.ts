import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";

const useHeaderStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: "30px 0",
      position: "relative",
    },
    nav: {
      "& a": {
        color: theme.palette.primary.main,
        textTransform: "uppercase",
        fontSize: "16px",
        transition: "all .3s ease",
        "&:hover": {
          color: theme.palette.secondary.main,
        },
      },
    },
  }),
);

export { useHeaderStyles };
