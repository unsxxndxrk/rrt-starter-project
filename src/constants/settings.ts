export const APP_DEFAULT_LANGUAGE: string = "en";
export const API_ENDPOINT: string =
  process.env.REACT_APP_API_ENDPOINT || window.location.origin;
