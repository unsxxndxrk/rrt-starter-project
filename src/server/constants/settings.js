const ORIGIN = "http://localhost";
const PORT = "8000";
const USERS_PATH = "./data/users.json";
const TODOS_PATH = "./data/todos.json";

exports.ORIGIN = ORIGIN;
exports.PORT = PORT;
exports.USERS_PATH = USERS_PATH;
exports.TODOS_PATH = TODOS_PATH;
