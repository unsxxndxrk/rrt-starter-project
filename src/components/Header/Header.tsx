import GeneralRoutes from "constants/generalRoutes";
import { useHeaderStyles } from "./Header.styles";
import { Link } from "react-router-dom";
import { Box, Grid } from "@material-ui/core";
import { Namespaces } from "i18n";
import { useTranslation } from "react-i18next";
import LanguageChanger from "./components/LanguageChanger/LanguageChanger";

const HeaderLayout = () => {
  const classes = useHeaderStyles();
  const { t } = useTranslation(Namespaces.Common);

  return (
    <Box className={classes.root}>
      <Grid
        container
        direction="row"
        justify="center"
        alignItems="center"
        spacing={5}
        className={classes.nav}
      >
        <Grid item>
          <Link to={GeneralRoutes.HOME}>{t("navigation.home")}</Link>
        </Grid>
        <Grid item>
          <Link to={GeneralRoutes.EXAMPLES}>{t("navigation.examples")}</Link>
        </Grid>
        <Grid item>
          <Link to={GeneralRoutes.ABOUT}>{t("navigation.about")}</Link>
        </Grid>
      </Grid>
      <LanguageChanger />
    </Box>
  );
};

export default HeaderLayout;
