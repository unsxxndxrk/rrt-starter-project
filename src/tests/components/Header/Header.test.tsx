import { render } from "tests/settings/render";
import Header from "components/Header/Header";

describe("<Header />", () => {
  test("should render Header component", () => {
    const wrapper = render(<Header />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
