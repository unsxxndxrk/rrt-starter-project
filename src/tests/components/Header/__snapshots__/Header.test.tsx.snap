// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header component 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root makeStyles-root"
  >
    <div
      class="MuiGrid-root makeStyles-nav MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <a
          href="/"
        >
          navigation.home
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <a
          href="/examples"
        >
          navigation.examples
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <a
          href="/about"
        >
          navigation.about
        </a>
      </div>
    </div>
    <div
      class="makeStyles-root"
    >
      <div
        class="makeStyles-langWrap"
      >
        <div
          class="makeStyles-lang active"
        >
          en
        </div>
        |
      </div>
      <div
        class="makeStyles-langWrap"
      >
        <div
          class="makeStyles-lang"
        >
          ru
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
